relay/pkg/block/forwarder.go

```diff
@@ -56,18 +64,59 @@ func RunForwarder(
 	loopCtx, cancelLoopCtx := context.WithCancel(ctx)
 
 	forwarder := &Forwarder{
-		btcChain:        btcChain,
-		hostChain:       hostChain,
-		headersQueue:    make(chan *btc.Header, headersQueueSize),
-		errChan:         make(chan error, 1),
-		loopExitHandler: cancelLoopCtx,
+		btcChain:                  btcChain,
+		hostChain:                 hostChain,
+		forwarderPullingSleepTime: defaultForwarderPullingSleepTime,
+		headersQueue:              make(chan *btc.Header, headersQueueSize),
+		errChan:                   make(chan error, 1),
+		loopExitHandler:           cancelLoopCtx,
 	}
 
+	go forwarder.pullingLoop(loopCtx)
 	go forwarder.pushingLoop(loopCtx)
 
 	return forwarder
 }
 
+func (f *Forwarder) pullingLoop(ctx context.Context) {
+	logger.Infof("running new block pulling loop")
+
+	defer func() {
+		logger.Infof("stopping current block pulling loop")
```

Comment:
Is it block pull or header pull?

Reply:
It originates from the approach taken in #767. I think we could leave as is and I will refactor that according to https://github.com/keep-network/tbtc/pull/767#discussion_r603913978

Reply:
Yeah, agreed. I wrote this comment before I realized what is happening and the discussion in https://github.com/keep-network/tbtc/pull/767#discussion_r603913978 😆